### PR TITLE
Issue #52: replace lodash with native/sublibs

### PR DESCRIPTION
**Pull request by mikeguta**
A solution for issue #52 (remove lodash as a dependency)
I chose to retain 2 of the `lodash` sublibs (`lodash.merge` and `lodash.groupby`) as PROD dependencies and I also kept `lodash.get` as a DEV dependency for tests (fewer casts needed)

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
     ],
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
+    "esModuleInterop": true,
```
Comment (epiphone) [inline tsconfig.json:17]: Why is this change needed?

**Review by epiphone — CHANGES_REQUESTED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -191,22 +200,29 @@ function getRequiredPropNames(
...
+    )
   }
   function isOptional(metas: ValidationMetadata[]) {
-    return _.some(metas, ({ type }) =>
-      _.includes([cv.ValidationTypes.CONDITIONAL_VALIDATION, cv.IS_EMPTY], type)
+    return (
+      metas &&
+      metas.some(
+        ({ type }) =>
+          [cv.ValidationTypes.CONDITIONAL_VALIDATION, cv.IS_EMPTY].indexOf(
```
[the remark this answers is not in the document]
Comment (mikeguta) [reply src/index.ts]: done

File: src/index.ts [hunk 3]
```diff
@@ -69,11 +79,8 @@ export function validationMetadatasToSchemas(userOptions?: Partial<IOptions>) {
 function getMetadatasFromStorage(
   storage: cv.MetadataStorage
 ): ValidationMetadata[] {
-  const metadatas: ValidationMetadata[] = _.get(storage, 'validationMetadatas')
-  const constraints: ConstraintMetadata[] = _.get(
-    storage,
-    'constraintMetadatas'
-  )
+  const metadatas: ValidationMetadata[] = (<any>storage).validationMetadatas
+  const constraints: ConstraintMetadata[] = (<any>storage).constraintMetadatas
```
[the remark this answers is not in the document]
Comment (mikeguta) [reply src/index.ts]: done

File: src/index.ts [hunk 4]
```diff
@@ -24,41 +25,50 @@ export function validationMetadatasToSchemas(userOptions?: Partial<IOptions>) {
...
-        value,
-        `target.${options.schemaNameField}`,
-        _.get(value, 'target.name')
-      )
+  const schemas: { [key: string]: SchemaObject } = {}
+  Object.entries(
+    _groupBy(
+      metadatas,
+      ({ target }) =>
+        (<any>target)[options.schemaNameField] ?? (<any>target).name
```
[the remark this answers is not in the document]
Comment (mikeguta) [reply src/index.ts]: perfect, done now b8dab97

File: tsconfig.json (hunk 1 above)
Comment (mikeguta) [reply tsconfig.json:17]: It's because of limitations in the import of modules where `exports = ...` syntax was used. You either get this error (for the default import)
image
Or you get this one (for the `*` variant)
image
In either case, the solution is the same, which is to turn on the `esModuleInterop` flag. With `tslib` being now referenced in most TS projects, there's little overhead to it anyway.

**Comment by mikeguta** [comment]
Hi @epiphone This should be ready for re-review.

File: tsconfig.json (hunk 1 above)
Comment (epiphone) [reply tsconfig.json:17]: I see, thanks :+1:

**Review by epiphone — APPROVED** [review]
:100:
